examples/yacat/yacat.py

```diff
@@ -0,0 +1,158 @@
+        min_mem_gib=0.5,
+        min_storage_gib=2.0,
+    )
+
+    async def worker_check_keyspace(ctx: WorkContext, tasks):
+        async for task in tasks:
+            keyspace_sh_filename = "keyspace.sh"
+            ctx.send_file(keyspace_sh_filename, "/golem/work/keyspace.sh")
+            ctx.run("/bin/sh", "/golem/work/keyspace.sh")
+            output_file = "keyspace.txt"
+            ctx.download_file("/golem/work/keyspace.txt", output_file)
+            yield ctx.commit()
+            task.accept_task()
+
+    async def worker_find_password(ctx: WorkContext, tasks):
+        ctx.send_file("in.hash", "/golem/work/in.hash")
+        # ctx.send_file(str(script_dir / "empty.txt"), "/golem/work/hashcat.potfile")
+
+        async for task in tasks:
+            skip = task.data
+            limit = skip + step
+
+            # Commands to be run on the provider
+            commands = (
+                "touch /golem/work/hashcat.potfile;"
+                f"hashcat -a 3 -m 400 /golem/work/in.hash --skip {skip} --limit {limit} {args.mask} -o /golem/work/hashcat.potfile"
+            )
+            ctx.run(f"/bin/sh", "-c", commands)
+
+            output_file = f"hashcat_{skip}.potfile"
+            ctx.download_file(f"/golem/work/hashcat.potfile", output_file)
+            yield ctx.commit()
+            task.accept_task(result=output_file)
+
+    # beginning of the main flow
+
+    write_hash(args.hash)
+    write_keyspace_check_script(args.mask)
+
+    # By passing `event_emitter=log_summary()` we enable summary logging.
+    # See the documentation of the `yapapi.log` module on how to set
+    # the level of detail and format of the logged information.
+    async with Engine(
+        package=package,
+        max_workers=args.number_of_providers,
+        budget=10.0,
+        # timeout should be keyspace / number of providers dependent
+        timeout=timedelta(minutes=25),
+        subnet_tag=args.subnet_tag,
+        event_emitter=log_summary(log_event_repr),
+    ) as engine:
+
+        # this is not typical use of engine.map as, there is only one task, with no data
+        async for task in engine.map(worker_check_keyspace, [Task(data=None)]):
+            pass
+
+        keyspace = read_keyspace()
+
+        print(
+            f"{utils.TEXT_COLOR_CYAN}"
+            f"Task computed: keyspace size count. The keyspace size is {keyspace}"
+            f"{utils.TEXT_COLOR_DEFAULT}"
+        )
+
+        step = int(keyspace / args.number_of_providers) + 1
+
+        ranges: range = range(0, keyspace, step)
+
+        async for task in engine.map(worker_find_password, [Task(data=range) for range in ranges]):
+            print(
+                f"{utils.TEXT_COLOR_CYAN}"
+                f"Task computed: {task}, result: {task.output}"
+                f"{utils.TEXT_COLOR_DEFAULT}"
+            )
+
+        password = read_password(ranges)
+
+        if password == None:
```

Comment:
```suggestion
        if password is None:
```
This is more "pythonic": https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_none.html